sensor/common/virtualmachine/service_impl.go

```diff
@@ -47,7 +50,9 @@ func (s *serviceImpl) UpsertVirtualMachine(ctx context.Context, req *sensor.Upse
 	}
 
 	log.Debugf("Upserting virtual machine: %s", req.VirtualMachine.GetId())
-	if err := s.component.Send(ctx, req.GetVirtualMachine()); err != nil {
+	timeoutCtx, cancel := context.WithTimeout(ctx, virtualMachineSendTimeout)
+	defer cancel()
+	if err := s.component.Send(timeoutCtx, req.GetVirtualMachine()); err != nil {
```

Comment:
Sth to check: Do we send `ResourceExhausted` when the context is canceled? Is the caller relying on us returning `ResourceExhausted` error when the sending should be retried?

In node-scanning, we decided to use acknowledgements when the message arrives to Central successfully, so it would be good to define the rules when transmission is retried.